rse-project.Rmd

```diff
@@ -21,7 +21,214 @@ small pieces in predictable places with readable names are easier to find and us
+whose primary aim is to produce tools rather than results,
+but is inappropriate for most data analysis work.
+The third tends to be too large:
+a team of half a dozen people may work on many different things at once,
+and a repository that holds them all quickly looks like someone's basement.
+
+The best rule of thumb we have found for deciding what is and isn't a project
+is to ask what you have meetings about.
+If the same set of people need to get together on a regular basis to talk about something,
+that "something" probably deserves its own repository.
+And if the list of people changes slowly over time but the meetings continue,
+that's an even stronger sign.
+
+## What files should every project contain? {#project-boilerplate}
+
+TODO: What are the bare minimums here?  I'd say must have `README`, even for your own projects that only live locally.  As soon as something is on GitHub must have `LICENSE` and `CONDUCT`?
+
+Most projects' repositories contain a few files that weren't mentioned in Noble's paper,
+but which have become conventional in open source projects.
+All of these files may be plain text or Markdown,
+and may have a `.txt` or `.md` suffix (or no suffix at all),
+but please use the principal names given,
+in all caps,
+since a growing number of tools expect them.
+
+-   `README`:
+    the project's title and a one-paragraph description of its purpose or content.
+    This file is displayed by GitHub Pages as the project's home page.
+
+-   `LICENSE`:
+    the project's license (discussed in Chapter \@ref(inclusive)).
+
+-   `CONDUCT`:
+    its code of conduct
+    (also discussed in Chapter \@ref(inclusive)).
+
+-   `CITATION`:
+    how the work should be cited.
+    This file usually contains a plain text citation,
+    and may also include entries formatted for various bibliographic systems like [BibTeX][bibtex].
+    Some projects also have a separate `CONTRIBUTORS` file listing everyone who has contributed to it,
+    while others include this as a section in `CITATION`.
+
+If setting up the project or contributing to it are complex,
+there may also be a file called `BUILD` that explains how to install the required software,
+the formatting conventions for new data entries,
+and so on.
+These instructions can also be included as a section in `README`.
+Whichever convention is used,
+remember that the easier it is for people to get set up and contribute,
+the more likely they are to do so @Stei2014.
+
+TODO: `README` or `CONTRIBUTING` are also great places to document the conventions of structure (the rest of this chapter), naming and style.
+
+## How do I document a project?
+
+TODO: What does someone need to know beyond the collection of files in the project? 
+ 
+TODO: What should go in a `README`? Use Elizabeth's Step 4: "Administrative and personnel details" and "Project information" prompts as a guide. https://www.ideals.illinois.edu/bitstream/handle/2142/91611/W2%20-%20DataDocumentationHandout.docx
```

Comment:
The mozilla open leaders modules have useful tips for this in their assignments for how to write readmes https://mozilla.github.io/open-leadership-training-series/articles/opening-your-project/write-a-great-project-readme/